src/src/Helpers/HTTPResponseParser.cpp

```diff
@@ -17,202 +17,191 @@
+                                                                                            eventName.c_str())) +
+                                                                      eventName.length() + 4;
+
+                                         // example( },"current":{"time":... ) we want to start after current":{
+
+                                         for (int i = 0; i < keyCount; i++) // Use keyCount to limit the iteration
+                                         {
+                                           String key = keys[i];
+                                           String value;
+                                           int    startIndex = str.indexOf(strformat(F("%s\":"), key.c_str()), startStringIndex);
 
-                                       String csv;
-                                       const int startStringIndex = str.indexOf(strformat(F("\"%s\":"),
-                                                                                          eventName.c_str())) +
-                                                                    eventName.length() + 4;
-
-                                       // example( },"current":{"time":... ) we want to start after current":{
-
-                                       for (int i = 0; i < keyCount; i++) // Use keyCount to limit the iteration
-                                       {
-                                         String key = keys[i];
-                                         String value;
-                                         int    startIndex = str.indexOf(strformat(F("%s\":"), key.c_str()), startStringIndex);
-
-                                         if (startIndex == -1) {
-                                           // Handle case where key is not found
-                                           value = F("-256"); // Placeholder value
-                                         } else {
-                                           int endIndex = 0;
-
-                                           if (!equals(eventName, F("current"))) {
-                                             // In daily and hourly, the values are stored in an array
-                                             startIndex += key.length() + 3; // Move index past the key
-                                             endIndex    = str.indexOf(']', startIndex);
+                                           if (startIndex == -1) {
+                                             // Handle case where key is not found
+                                             value = F("-256"); // Placeholder value
                                            } else {
-                                             startIndex += key.length() + 2; // Move index past the key
-                                             endIndex    = str.indexOf(',', startIndex);
+                                             int endIndex = 0;
+
+                                             if (!equals(eventName, F("current"))) {
```

Comment:
I don't understand the next few lines as you offset `startIndex` with a fixed offset of 3 or 2.
And then try to match with `]` or `}`.
So I guess there should be some `[` or `{`, right?

Maybe you can add some example strings in comments to help understand these lines?